src/wp-includes/blocks.php

```diff
@@ -2713,18 +2714,59 @@ function build_query_vars_from_query_block( $block, $page ) {
 			}
 			$query['tax_query'] = array_merge( $query['tax_query'], $tax_query_back_compat );
 		}
-		if ( ! empty( $block->context['query']['taxQuery'] ) ) {
-			$tax_query = array();
-			foreach ( $block->context['query']['taxQuery'] as $taxonomy => $terms ) {
-				if ( is_taxonomy_viewable( $taxonomy ) && ! empty( $terms ) ) {
-					$tax_query[] = array(
-						'taxonomy'         => $taxonomy,
-						'terms'            => array_filter( array_map( 'intval', $terms ) ),
-						'include_children' => false,
-					);
+
+		if ( ! empty( $block->context['query']['taxQuery'] ) && is_array( $block->context['query']['taxQuery'] ) ) {
+			$tax_query_input = $block->context['query']['taxQuery'];
+			$tax_query       = array();
+			// If there are keys other than include/exclude, it's the old
+			// format e.g. "taxQuery":{"category":[4]}
+			if ( ! empty( array_diff( array_keys( $tax_query_input ), array( 'include', 'exclude' ) ) ) ) {
+				foreach ( $block->context['query']['taxQuery'] as $taxonomy => $terms ) {
+					if ( is_taxonomy_viewable( $taxonomy ) && ! empty( $terms ) ) {
+						$tax_query[] = array(
+							'taxonomy'         => $taxonomy,
+							'terms'            => array_filter( array_map( 'intval', $terms ) ),
+							'include_children' => false,
+						);
+					}
 				}
+			} else {
+				// This is the new format e.g. "taxQuery":{"include":{"category":[4]},"exclude":{"post_tag":[5]}}
+
+				// Helper function to build tax_query conditions from taxonomy terms.
+				$build_conditions = static function ( array $terms, string $operator = 'IN' ): array {
+					$conditions = array();
+					foreach ( $terms as $taxonomy => $terms ) {
+						if ( ! empty( $terms ) && is_taxonomy_viewable( $taxonomy ) ) {
+							$conditions[] = array(
+								'taxonomy'         => $taxonomy,
+								'terms'            => array_filter( array_map( 'intval', $terms ) ),
+								'operator'         => $operator,
+								'include_children' => false,
+							);
+						}
+					}
```

Comment:
Cheers to PhpStorm for flagging this variable naming conflict, though I realize it may not impact the execution:

<img width="773" height="263" alt="Image" src="https://github.com/user-attachments/assets/c93e827a-0420-4044-8872-652939731c1a" />

```suggestion
					foreach ( $terms as $taxonomy => $tax_terms ) {
						if ( ! empty( $tax_terms ) && is_taxonomy_viewable( $taxonomy ) ) {
							$conditions[] = array(
								'taxonomy'         => $taxonomy,
								'terms'            => array_filter( array_map( 'intval', $tax_terms ) ),
								'operator'         => $operator,
								'include_children' => false,
							);
						}
					}
```

Reply:
Good catch! It seems there is no actual problem due to internals of `foreach`, but let's definitely update this.